File: src/test/testData/java/LoopStatements.java
class Tests {
    @Complexity(4)
    public void allPossibleLoops() {
        while (true) {           // +1
            a++;
        }

        do {                     // +1
            a++;
        } while (true)

        for (int i = 0; i++; i < 10) { // +1
            a++;
        }

        for (int i : counts) {    // +1
            a++;
        }

        List<String> tokens = new ArrayList();
        tokens.stream()
              .forEach(it -> a++);
    }

    @Complexity(5)
    public void loopsCreateNesting() {
        while (true) {            // +1
            if (true) {           // +2 (nesting = 1)
                a++
            } else {              // +1

            }
        }

        for (int i : counts) {    // +1
            a++;
        }
    }

    @Complexity(2)
    public void lambdaAddsNestingOnly() {
        List<String> tokens = new ArrayList();
        tokens.stream()
              .filter(it -> it != null)
              .forEach(it -> {         // nesting = 1
                  if (true) {          // +2 (nesting = 1)
                      a++
                  }
              });
    }
}

File: src/test/testData/dart/LoopStatements.dart
class Tests {
  int a = 0;
  List<int> counts = [];

  @complexity(4)
  void allPossibleLoops() {
    while (true) {           // +1
      a++;
    }

    do {                     // +1
      a++;
    } while (true);

    for (int i = 0; i < 10; i++) {     // +1
      a++;
    }

    for (final i in counts) {          // +1
      a++;
    }

    final tokens = <String>[];
    tokens.forEach((it) => a++);       // closure: +nesting only, no +complexity
  }

  @complexity(5)
  void loopsCreateNesting() {
    while (true) {           // +1
      if (true) {            // +2 (nesting = 1)
        a++;
      } else {               // +1
      }
    }

    for (final i in counts) {  // +1
      a++;
    }
  }

  @complexity(2)
  void lambdaAddsNestingOnly() {
    final tokens = <String>[];
    tokens.where((it) => it.isNotEmpty).forEach((it) {  // nesting = 1
      if (true) {             // +2 (nesting = 1)
        a++;
      }
    });
  }
}
